projects/Unstructured Supplemenrary  Service Data/ussdtim.py

```diff
@@ -0,0 +1,194 @@
+	temp_storage.extend(s_name)
+	temp_storage.extend(sex)
+	temp_storage.extend(gen_bvn)
+	details= " ".join(temp_storage)
+	print(details)
+	print("1. Press # to go back to options menu\n2. Press * to exit")
+	bck=input(":")
+	if bck=='#':
+	    options_menu( )
+	else:
+	    sys.exit( )
+	
+	exit( )
+	
+def upgrade_migrate( ):
+		print("Welcome to our online Upgrade/Migration services.\n 1. Ugrade\n 2. Migrate")
+		print("press # is go back to the Main Menu.")
+		prompt = input("Enter preferred Choice:")
+		if prompt=="1":
+			time.sleep(5)
+			print("Upgrading...")
+			exit( )
+		elif prompt == "2":
+			time.sleep(5)
+			print("Migrating...")
+			exit( )
+		elif prompt == "#":
+			options_menu( )
+		else:
+			sys.exit( )
+
+def balance ( ):
+	print("ACCOUNT\tBALANCE\n CHECKER")
+	print("press # is go back to the Main Menu.")
+	pin=input("Enter your 4 digit pin:")
+# isdigit( ) is used to  check for digits within a str while the nested if is used to make sure the user inputs 4 digits.
+
+###```i am to put the pin trial in a while loop```###REMINDER!!!
+
+	if len(pin)!=4:
+	   	print("Make sure its a 4digit pin.")
+	   	time.sleep(5)
+	   	balance( )
+	else:
+	   		    if pin.isdigit( ):
+	   		        time.sleep(5)
+	   		        print("Loading...")
+	   		        exit( )
+	   		    elif pin== "#":
+	   		        options_menu( )
+	   		    else:
+	   		        time.sleep(15)
+	   		        print("wrong pin")
+	   		        sys.exit( )
+		
+def transf( ):
+	print("1. Transfer self\n2. Transfer others")
+	print("press # is go back to the Main Menu.")
+	trnsf=input(":")
+	if trnsf == "#" :
+	    options_menu( )
+	elif trnsf == "1":
+	    time.sleep(5)
+	    print("Sending...")
+	    exit( )
+	elif trnsf=="2":
+	    time.sleep(5)
+	    num=int(input("Enter receivers mobile number:"))
+	    print("Transferring to",num)
+	    exit( )
+	else:
+	    if trnsf.isdigit( )!= True:
+	        time.sleep(5)
+	        print("Not an option")
+	        sys.exit( )
+	    elif trnsf.isdigit( ) and len(trnsf)>2:
+	        time.sleep( 5)
+	        print("wrong password.")
+	        sys.exit( )
+	    else:
+	        time.sleep(10)
+	        print("An error has occurred")
+	        sys.exit( )
+	        	
+def funds( ):
+	time.sleep(3)
+	print(bank_list)
+	bnk = input("Select receipients Bank:")
+	acc_num= input("Entet account number:")
+	print("Sending to",acc_num)
+	hash= input("1.Press # to go back to options menu\n2. Press * to go exit.")
+	if hash == "#":
+	   options_menu( )
+	elif hash == "*":
+	    exit( )
+	else:
+														sys.exit( )											
+
+																#--------------------------------------------------												
+###i'm yet to catch an error for non -digit and more than one digit###REMINDER!!!						#-#------------------------------------------------------								
+# This is the function for options.
+def options_menu( ) :
+	print("1. Open Account\n2. Upgrade/Migrate\n3. Balance\n4. Transfer\n5. Funds")
+	choice=input("Enter an option:")

```

Comment:
```python
	choice=input("Enter an option:")
	if choice == "1":
		time.sleep(10)
		open_acct( )
	elif choice == "2":
		time.sleep(10)
		upgrade_migrate( )
	elif choice =="3":
		time.sleep(10)
		balance( )
	elif choice =="4":
		time.sleep(10)
		transf( )
	elif choice =="5":
		time.sleep(10)
		funds( )
	else:
		time.sleep(5)
		print("Not an option.")
		sys.exit( )
```

Alternative to  write multiple if else conditions
```python

select_options ={
	  '1':open_acct,
	  '2':upgrade_migrate,
	  '3': balance,
	  '4':transf,
	  '5':funds
}
choice=input("Enter an option:")
if select_options.get(choice):
  select_options[choice]()
else:
  sys.exit()

```

Reply:
Ok will do this 
Thanks for the enlightenment